lib/core/include/irods/base64.h

```diff
@@ -11,17 +11,12 @@
 
 #include "irods/rodsDef.h"
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
-int base64_encode( const unsigned char *in,  unsigned long inlen,
-                   unsigned char *out, unsigned long *outlen );
-int base64_decode( const unsigned char *in,  unsigned long inlen,
-                   unsigned char *out, unsigned long *outlen );
-
-#ifdef __cplusplus
-}
-#endif
+namespace irods
+{
+    int base64_encode( const unsigned char *in,  unsigned long inlen,
+                       unsigned char *out, unsigned long *outlen );
+    int base64_decode( const unsigned char *in,  unsigned long inlen,
+                       unsigned char *out, unsigned long *outlen );
+} // namespace irods
```

Comment:
Since we are removing the C linkage, should we rename this file to base64.hpp to further indicate that C clients should not be using it?

Reply:
Given that this change breaks anyone originally relying on these functions, it is okay to change the file extension.

After saying that, we need to think about this a bit more.

Given that the dev pkg exposes almost everything used in iRODS, we can't break people's code. We need to provide a solution that let's the user transition gracefully or we need to define a list of things people are allowed to use.

Below is a potential plan for the transitional approach.
- Leave the old functions in base64.h and mark them deprecated
- Create a new file with the .hpp extension and copy @JustinKyleJames's new version of base64.h into it
- Add a `#warning` to the base64.h file that let's devs know the header is deprecated and that they shouldn't use it anymore. The warning could point the dev to the cpp file or maybe we don't ship the hpp version at all

Reply:
Are clients using this directly?  I have my doubts whether they are.

The Globus client only used it indirectly via the clientLogin() call.

Reply:
The requested changes have been implemented.